### PR TITLE
ENYO-426: Marquee shows ellipsis but Does Not start rolling in moon.InputHeader Enyo-DCO-1.1-Signed-off-by: Artem Krasnyi <krasnyi.artem@lge.com>

### PR DESCRIPTION
**Pull request by akrasnyi**
…nputHeader

### CODE REVIEW

File: lib/Header/Header.less [hunk 1]
```diff
@@ -182,4 +182,12 @@
...
 		}
 	}
+}
+
+.hidden {
+  display: none!important;
+}
+
+.moon-input-header .moon-input-header-input-decorator .moon-input.header-marquee {
+  line-height: 1.24em;
```
Comment (aarontam) [inline lib/Header/Header.less]: Where did this number come from? I noticed that when switching between the Input and the Marquee, there was a noticeable shift in the positioning of the controls.

File: lib/Header/Header.js [hunk 2]
```diff
@@ -796,9 +797,31 @@ module.exports = kind(
...
+
+        /**
+         * Logic for replacing input into marquee
+         *
+         * @private
+         */
+
+        marqueToggle: function (){
+            if (this.$.titleInputMarquee.getContent() === '') return ;
+            if (this.$.titleInput.hasClass('hidden')){
```
Comment (aarontam) [inline lib/Header/Header.js]: Can't we simplify this to the following (and remove the `inputBlur` handler`)?

```
var isInputShowing = this.$.titleInput.get('showing');
this.$.titleInput.set('showing', !isInputShowing);
this.$.titleInputMarquee.set('showing', isInputShowing);
```

File: lib/Header/Header.js [hunk 3]
```diff
@@ -796,9 +797,31 @@ module.exports = kind(
...
+        this.marqueToggle();
+    },
+
+        /**
+         * Logic for replacing input into marquee
+         *
+         * @private
+         */
+
+        marqueToggle: function (){
```
Comment (aarontam) [inline lib/Header/Header.js]: This should be `marqueeToggle`.

File: lib/Header/Header.js [hunk 4]
```diff
@@ -305,7 +305,8 @@ module.exports = kind(
 		{name: 'titleWrapper', kind: Control, classes: 'moon-header-title-wrapper', components: [
 			{name: 'title', kind: MarqueeText, classes: 'moon-header-text moon-header-title', canGenerate: false},
 			{name: 'inputDecorator', kind: InputDecorator, classes: 'moon-input-header-input-decorator',canGenerate: false, components: [
-				{name: 'titleInput', kind: Input, classes: 'moon-header-text moon-header-title'}
+                {name: 'titleInput', kind: Input, classes: 'moon-header-text moon-header-title',  onblur: 'inputBlur'},
+                {name: 'titleInputMarquee', kind: MarqueeText, classes: 'moon-header-text moon-header-title moon-input header-marquee hidden', marqueeOnRender: true, ontap: "marqueToggle"}
```
Comment (aarontam) [inline lib/Header/Header.js]: There's no need for the `hidden` class as we have a `showing` property that can be set for Controls.

File: lib/Header/Header.js [hunk 5]
```diff
@@ -796,9 +797,31 @@ module.exports = kind(
 	* @fires module:moonstone/Header~Header#onInputHeaderChange
 	* @private
 	*/
-	handleChange: function (inSender, inEvent) {
-		this.doInputHeaderChange({originalEvent: util.clone(inEvent, true)});
-	},
+    handleChange: function (inSender, inEvent) {
+        this.doInputHeaderChange({originalEvent: util.clone(inEvent, true)});
+        this.$.titleInputMarquee.setContent(event.originator.getValue());
```
Comment (aarontam) [inline lib/Header/Header.js]: Let's use the generic setter here i.e.

```
this.$.titleInputMarquee.set('content', event.originator.getValue());
```

File: lib/Header/Header.js [hunk 6]
```diff
@@ -796,9 +797,31 @@ module.exports = kind(
...
+
+        marqueToggle: function (){
+            if (this.$.titleInputMarquee.getContent() === '') return ;
+            if (this.$.titleInput.hasClass('hidden')){
+                this.$.titleInput.removeClass('hidden');
+                this.$.titleInputMarquee.addClass('hidden');
+            }
+        },
+
+        inputBlur: function () {
```
Comment (aarontam) [inline lib/Header/Header.js]: This can probably be removed, based on the above comment.

File: lib/Header/Header.js (hunk 5 above)
Comment (aarontam) [inline lib/Header/Header.js]: This should also reference the event parameter, which is `inEvent` in this case.

File: lib/Header/Header.js (hunk 4 above)
Comment (aarontam) [inline lib/Header/Header.js]: Also, do we actually need to set `marqueeOnRender:true` here? And we should surround the tap handler with single quotes, not double quotes.

File: lib/Header/Header.js [hunk 7]
```diff
@@ -305,7 +305,8 @@ module.exports = kind(
 		{name: 'titleWrapper', kind: Control, classes: 'moon-header-title-wrapper', components: [
 			{name: 'title', kind: MarqueeText, classes: 'moon-header-text moon-header-title', canGenerate: false},
 			{name: 'inputDecorator', kind: InputDecorator, classes: 'moon-input-header-input-decorator',canGenerate: false, components: [
-				{name: 'titleInput', kind: Input, classes: 'moon-header-text moon-header-title'}
+                {name: 'titleInput', kind: Input, classes: 'moon-header-text moon-header-title',  onblur: 'inputBlur'},
```
Comment (aarontam) [inline lib/Header/Header.js]: We generally use tabs for indentation, not spaces - you might need to change the settings in your editor.

File: lib/Header/Header.js [hunk 8]
```diff
@@ -796,9 +797,31 @@ module.exports = kind(
...
+    },
+
+        /**
+         * Logic for replacing input into marquee
+         *
+         * @private
+         */
+
+        marqueToggle: function (){
+            if (this.$.titleInputMarquee.getContent() === '') return ;
```
Comment (aarontam) [inline lib/Header/Header.js]: Let's use the generic getter here i.e.

```
if (this.$.titleInputMarquee.get('content') === '') return;
```

File: lib/Header/Header.js [hunk 9]
```diff
@@ -796,9 +797,25 @@ module.exports = kind(
 	* @fires module:moonstone/Header~Header#onInputHeaderChange
 	* @private
 	*/
-	handleChange: function (inSender, inEvent) {
-		this.doInputHeaderChange({originalEvent: util.clone(inEvent, true)});
-	},
+    handleChange: function (inSender, inEvent) {
+        this.doInputHeaderChange({originalEvent: util.clone(inEvent, true)});
+        this.$.titleInputMarquee.set('content', inEvent.originator.getValue());
+        this.marqueToggle();
```
Comment (aarontam) [inline lib/Header/Header.js]: This should be `marqueeToggle` :smile:

File: src/Header/Header.less [hunk 10]
```diff
@@ -237,3 +237,7 @@
 		float: right;
 	}
 }
+
+.hidden {
```
Comment (clinuz) [inline src/Header/Header.less]: We can't have a global `.hidden` class defined here. This doesn't make sense.

**Comment by lishammel** [comment]
Closing PR as Jira has been closed.

**Pull request closed without merging**